src/System.Text.Json/tests/Serialization/PolymorphicTests.cs

```diff
@@ -130,6 +130,13 @@ public static void ArrayAsRootObject()
             json = JsonSerializer.ToString<object>(ireadonlylist);
             Assert.Equal(ExpectedJson, json);
 
+            ISet<object> iset = new HashSet<object> { 1, true, address, null, "foo" };
+            json = JsonSerializer.ToString(iset);
+            Assert.Equal(ExpectedJson, json);
```

Comment:
Order isn't guaranteed, right? Looks like the tests for `ImmutableSet` which use `expectedObjects` are a better way to test.

(Same for HashSet tests in this file, which I realize you did not add.)